proto/recipe_specification.proto

```diff
@@ -18,8 +19,11 @@ message RecipeSchema {
   // What resources this plugin can handle with their parameter constraints
   repeated ResourcePattern supported_resources = 4;
   
+  // What scheduling capabilities this plugin supports
```

Comment:
Let's add a ScheduleVersion so we can move to a more expressive schedule in the future